bin/utils/component-utils.sh

```diff
@@ -503,3 +503,175 @@ process_component_appfw_plugin() {
+# - NODE_HOME
+# - ROOT_DIR
+#
+# Example:
+# - List the desktop plugin IDs for the explorer-mvs component
+#   i.e: list_component_plugin_id "${ROOT_DIR}/components/explorer-mvs"
+#
+# @param string   component directory
+
+list_component_plugin_id() {
+  component_dir=$1
+
+  definition_file="pluginDefinition.json"
+
+  cd ${component_dir}
+
+  appfwplugin_index=0
+  appfwplugin_definition_file=$(read_component_manifest "${component_dir}" ".appfwPlugins[${appfwplugin_index}].path" 2>/dev/null)
+  while [ "${appfwplugin_definition_file}" != "null" ] && [ -n "${appfwplugin_definition_file}" ]; do
+    echo $(read_json "${appfwplugin_definition_file}/${definition_file}" ".identifier" 2>/dev/null)
+    appfwplugin_index=`expr $appfwplugin_index + 1`
+    appfwplugin_definition_file=$(read_component_manifest "${component_dir}" ".appfwPlugins[${appfwplugin_index}].path" 2>/dev/null)
+  done
+  desktopIframe_index=0
+  desktopIframe_id=$(read_component_manifest "${component_dir}" ".desktopIframePlugins[${desktopIframe_index}].id" 2>/dev/null)
+  while [ "${desktopIframe_id}" != "null" ] && [ -n "${desktopIframe_id}" ]; do
+    echo ${desktopIframe_id}
+    desktopIframe_index=`expr $desktopIframe_index + 1`
+    desktopIframe_id=$(read_component_manifest "${component_dir}" ".desktopIframePlugins[${desktopIframe_index}].id" 2>/dev/null)
+  done
+  return 0
+}
+
+###############################
+# Verifies a specific component by performing a fast check. If the component
+# has any services, it will check if the status of those services are "UP". If
+# it has any desktop plugins it will check if it exists on the zowe instance.
+#
+# Note: this function calls is dependent on various utility functions and
+#       environment variables. Simply source bin/internal/prepare-environment.sh
+#       to import the required dependencies.
+#
+# Required environment variables:
+# - NODE_HOME
+# - ROOT_DIR
+# - ZOWE_EXPLORER_HOST
+# - DISCOVERY_PORT
+# - ZOWE_ZLUX_SERVER_HTTPS_PORT
+#
+# Example:
+# - Verify service and desktop plugin of explorer-jes component
+#   i.e: verify_component_instance "${ROOT_DIR}/components/explorer-jes"
+#
+# @param string   component directory
+verify_component_instance() {
+  component_id=$1
+
+  component_dir=$(find_component_directory "${component_id}")
+
+  service_ids=$(list_component_service_id "${component_dir}")
+
+  if [ ! -d ~/temp-verify ]; then
+    mkdir ~/temp-verify
+  fi
+
+  echo ${component_id} | tr '[:lower:]' '[:upper:]'
+  echo "==========================================\n"
+
+  echo "SERVICE COMPONENT STATUS"
+
+  for service_id in $service_ids; do
+    rm -rf ~/temp-verify/verify-ext-serviceid.json
+    node "${ROOT_DIR}"/bin/utils/curl.js https://"${ZOWE_EXPLORER_HOST}":"${DISCOVERY_PORT}"/eureka/apps/"${service_id}" -k -H 'Accept: application/json' -J >> ~/temp-verify/verify-ext-serviceid.json
+    service_status=$(read_json ~/temp-verify/verify-ext-serviceid.json .application.instance[0].status)
```

Comment:
let's try if this works:

1. pipe: `node curl.js url | read_json '.application.instance[0].status'`
2. redirect input/output: `read_json '.application.instance[0].status' < node curl.js url`

Reply:
Purpose is to avoid creating tmp files

Reply:
Made changes to resolve this issue